code/__HELPERS/game.dm

```diff
@@ -82,7 +82,7 @@
 /proc/get_alien_candidate()
 	var/mob/dead/observer/picked
 
-	for(var/i in GLOB.observer_list)
+	for(var/i in shuffle(GLOB.observer_list))
```

Comment:
The observer_list is added to and removed each time someone logs in rather than turns into a ghost. So the first person on the list is the first to log in. That means the older users get first chance for being a xeno (passing checks rather than it being the oldest death) so being truly random is more fair.